integrated_channels/api/v1/canvas/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+Viewsets for Canvas
+"""
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from edx_rbac.decorators import permission_required
+from enterprise.models import EnterpriseCustomerUser
+
+
+from integrated_channels.canvas.models import CanvasEnterpriseCustomerConfiguration
+
+from .serializers import CanvasEnterpriseCustomerConfigurationSerializer
+
+
+class CanvasConfigurationViewSet(viewsets.ModelViewSet):
+    serializer_class = CanvasEnterpriseCustomerConfigurationSerializer
+    permission_classes = (permissions.IsAuthenticated, permissions.DjangoModelPermissions,)
+
+    def get_queryset(self):
+        if self.request.user.is_staff:
+            return CanvasEnterpriseCustomerConfiguration.objects.all()
+        try:
+            enterprise_customer = EnterpriseCustomerUser.objects.get(user_id=self.request.user.id).enterprise_customer
```

Comment:
do we typically send a user id or how might that look in the front end? I know we track enterprise customer id already which is why I just went with passing that along.

I think practically there is probably no one who belongs to multiple enterprises and also isn't a staff user, but that is a possible gap here as well.

Reply:
Also, if we use the retrieve endpoint on the frontend, does retrieve expect an enterprise customer? Judging by "pk" I'm guessing it's looking for a pure id that we wouldn't have available on the front end (unless we just created the config), so the only endpoint usable would be "list" and the front ends for most (if not all) of the sidebar objects assumes/(passes to the api) 1 enterprise customer.

Reply:
> do we typically send a user id or how might that look in the front end?

Should always have a User ID for an authenticated learner who's requesting data. Believe it gets unmarshalled out from the jwt. 

Good call on the multiple enterprises though, should handle this via a filter instead of get just in case multiple are returned.

Retrieve doesn't expect an enterprise customer and it shouldn't, though we should be able to support querystring args for querying.

Right now it expects the primary key for the Conf object. Since we're accessing the view directly PK made sense. For getting to a point of displaying which views are available, I figured the typical approach is "Hit the list endpoint to see what options are available, dive in via the retrieve endpoint".

Reply:
I think the list to specific flow makes sense generally, but I'm not sure it's really suitable here.
We shouldn't have multiple configs so we should be going to directly to a specific result.

We could still use a list and just assume the first item of the list is the result we want, but yeah I'm still iffy on parameters there because we're currently allowing multiple enterprises on the backend. That was why we went with a query string enterprise customer, so that even ECS/multi-enterprise users could limit down to one customer.

Reply:
Would it make more sense to just serialize this data as part of the EnterpriseCustomer endpoint and include all of configs as part of the response body?

```
{
	"uuid": "b0990312-c5a3-48c4-b0fe-b8d844249623",
	"name": "TestDemoEnterpriseTest",
	"slug": "testdemoenterprisetest",
	"active": true,
	"site": {
		"domain": "courses.stage.edx.org",
		"name": "stage"
	},
	"enable_data_sharing_consent": true,
	"enforce_data_sharing_consent": "at_enrollment",
	"branding_configuration": {
		"enterprise_customer": "b0990312-c5a3-48c4-b0fe-b8d844249623",
		"enterprise_slug": "testdemoenterprisetest",
		"logo": "https://edxuploads.s3.amazonaws.com/enterprise/branding/1/1_logo.png",
		"primary_color": "#1a337b",
		"secondary_color": "#d7e3fc",
		"tertiary_color": "#007d88"
	},
	"identity_provider": "saml-arbi-test-shib",
	"enable_audit_enrollment": true,
	"replace_sensitive_sso_username": false,
	"enable_portal_code_management_screen": false,
	"sync_learner_profile_data": true,
	"enable_audit_data_reporting": true,
	"enable_learner_portal": true,
	"enable_portal_reporting_config_screen": false,
	"enable_portal_saml_configuration_screen": false,
	"contact_email": null,
	"enable_portal_subscription_management_screen": false,
	"hide_course_original_price": false,
	"enable_analytics_screen": false,
	"enable_integrated_customer_learner_portal_search": true
}
```

and add in

```
moodle_lms_configuration: {},
canvas_lms_configuration: {},
```

What I want to avoid is us having something "smart" that'll just parse out and return whatever is on the backend like ```third_party_lms_config: {type: 'moodle'}```, where you're constantly trying to figure out what the type is on the frontend.

Reply:
This seems like it just changes at which point we do the API calls without really changing the backend logic to me unless I'm missing something?

If you're proposing a foreign key relationship on the enterprise customer <> every configuration table that seems a bit heavy.
If you're proposing making the API call here during the EnterpriseCustomer view/api call then that's fine, but I am not sure that really changes the issue of returning all enterprise configurations vs only 1.

My only issue here was that we need to presume only 1 enterprise customer and the permission logic of checking if a user is staff and returning all configurations was counter to what we expect since it can result in many, many configurations.